filters/profilter.go

```diff
@@ -21,29 +21,12 @@ type LanternProFilter struct {
 	proTokens *set.Set
 }
 
-type optSetter func(f *LanternProFilter) error
-
-func Logger(l utils.Logger) optSetter {
-	return func(f *LanternProFilter) error {
-		f.log = l
-		return nil
-	}
-}
-
-func New(next http.Handler, setters ...optSetter) (*LanternProFilter, error) {
-	f := &LanternProFilter{
-		log:       utils.NullLogger,
+func NewProFilter(next http.Handler, log utils.Logger) *LanternProFilter {
+	return &LanternProFilter{
+		log:       log,
 		next:      next,
 		proTokens: set.New(),
 	}
-
-	for _, s := range setters {
-		if err := s(f); err != nil {
-			return nil, err
-		}
-	}
-
-	return f, nil
```

Comment:
The setters pattern is used by Oxy. Even though we finally don't use Oxy, I find it very useful for extending filters with optional and configurable options. You are just removing this possibility with this simplification.